pages/applications/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { ApplicationStageType, RoleType } from '@prisma/client';
+import { Col, Form, Row, Spin, Table } from 'antd';
+import Search from 'antd/lib/input/Search';
+import Title from 'antd/lib/typography/Title';
+import { ChangeEventHandler, useState } from 'react';
+import useSWR from 'swr';
+import api from '../../api/api';
+import { APPLICATIONS_API_ENDPOINT } from '../../api/applicationsApi';
+import ApplicationStagesSelect from '../../components/forms/ApplicationStagesSelect';
+import RoleTypesSelect from '../../components/forms/RoleTypesSelect';
+import { ApiResponse } from '../../types/apiResponse';
+import { ApplicationListData, ApplicationQueryParams } from '../../types/application';
+import { splitByWhitespaces } from '../../utils/strings/formatters';
+
+function Applications() {
+  const [searchParams, setSearchParams] = useState<ApplicationQueryParams>({
+    searchWords: [],
+    roleTypeWords: [],
+    stageTypeWords: [],
+  });
+
+  const onSearchBarChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    setSearchParams({ ...searchParams, searchWords: splitByWhitespaces(e.target.value) });
+  };
+
+  const onRoleTypesFilterChange = (roleTypes: RoleType[]) => {
+    setSearchParams({ ...searchParams, roleTypeWords: roleTypes });
+  };
+
+  const onApplicationStageTypesFilterChange = (stageTypes: ApplicationStageType[]) => {
+    setSearchParams({ ...searchParams, stageTypeWords: stageTypes });
+  };
+
+  const { data } = useSWR<ApiResponse<ApplicationListData[]>>(
+    [APPLICATIONS_API_ENDPOINT, searchParams],
+    (url, searchParams) => api.get(url, { params: searchParams }),
+  );
```

Comment:
Are we able to debounce this or throttle this? I think throttle would be better here.